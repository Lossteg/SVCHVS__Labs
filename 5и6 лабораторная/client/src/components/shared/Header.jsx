import React, { useEffect, useState } from 'react';
import { getUserData } from '../../api/userAPI';
import { useTheme } from '@mui/system';
import { AppBar, Toolbar, Container, Typography, Box, IconButton, Avatar, Tooltip } from '@mui/material';
import Logo from "./Logo";
import UserAvatar from "../../assets/UserAvatar.png";
import { useNavigate } from 'react-router-dom';

function Header() {
  const navigate = useNavigate();
  const [userData, setUserData] = useState({});
  const theme = useTheme();

  useEffect(() => {
    const fetchUserData = async () => {
      try {
        const data = await getUserData();
        setUserData(data);
      } catch (error) {
        console.error("Error in Header component:", error);
      }
    };
    fetchUserData();
  }, []);

  const handleLogoClick = () => {
    console.log("Я кликнул на лого")
    navigate("/");
  }

  const handleAvatarClick = () => {
    console.log("Я кликнул на аватарку")
    navigate("/profile");
  }

  return (
    <AppBar component="header" position="fixed">
      <Container maxWidth="xl" sx={{ pt: "0.3rem", pb: "0.3rem" }}>
        <Toolbar
          disableGutters
          sx={{ justifyContent: "space-between", padding: 0 }}
        >
          <Logo className="logo-header" onClick={handleLogoClick}/>
          <Box
            sx={{
              display: "flex",
              flexDirection: "column-reverse",
              alignItems: "center",
              [theme.breakpoints.down("600")]: {
                flexDirection: "row",
                gap: "0.4rem",
              },
            }}
          >
            <Typography
              className="welcome-user"
              sx={{
                fontSize: "1.1rem",
                [theme.breakpoints.down("1100")]: {
                  fontSize: "1rem",
                },
                [theme.breakpoints.down("md")]: {
                  fontSize: "0.9rem",
                },
                [theme.breakpoints.down("sm")]: {
                  fontSize: "0.8rem",
                },
              }}
            >
              Welcome, {userData.name}!
            </Typography>
            <Tooltip title="My Orders" placement="left">
              <IconButton sx={{ p: 0, backgroundColor: "white" }} onClick={handleAvatarClick}>
                <Avatar
                  alt="User-avatar"
                  src={UserAvatar}
                  sx={{
                    width: "3vw",
                    [theme.breakpoints.down("1350")]: {
                      width: "3.5vw",
                    },
                    [theme.breakpoints.down("1100")]: {
                      width: "3.8vw",
                    },
                    [theme.breakpoints.down("md")]: {
                      width: "4.3vw",
                    },
                    [theme.breakpoints.down("768")]: {
                      width: "4.8vw",
                    },
                    [theme.breakpoints.down("600")]: {
                      width: "5.9vw",
                    },
                    [theme.breakpoints.down("520")]: {
                      width: "6.6vw",
                    },
                    [theme.breakpoints.down("400")]: {
                      width: "9vw",
                    },
                    height: "auto",
                  }}
                />
              </IconButton>
            </Tooltip>
          </Box>
        </Toolbar>
      </Container>
    </AppBar>
  );
}

export default Header;