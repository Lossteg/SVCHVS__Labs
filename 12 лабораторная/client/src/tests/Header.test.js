import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../components/shared/Header';

describe('Header component', () => {
  it('should navigate to the home page on logo click', async () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    const headerComponent = container.querySelector('.logo-header');

    fireEvent.click(headerComponent);

    expect(window.location.pathname).toBe('/');
  });
  it('loads the user avatar image', async () => {
    const { getByAltText } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    // Ждем, пока изображение загрузится (может потребоваться время)
    const userAvatar = await waitFor(() => getByAltText('User-avatar'));

    expect(userAvatar).toBeTruthy();
  });
  it('Header component matches snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });
});

