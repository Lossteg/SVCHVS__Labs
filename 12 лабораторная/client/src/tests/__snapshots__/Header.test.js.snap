// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header component matches snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-gk3fd8-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular css-u2rs4q-MuiToolbar-root"
    >
      <img
        alt="Copy-center-logo"
        className="logo-header"
        onClick={[Function]}
        src="CopyCenterLogo.png"
      />
      <div
        className="MuiBox-root css-1fv2ldg"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 welcome-user css-ofhtmv-MuiTypography-root"
        >
          Welcome, 
          !
        </p>
        <button
          aria-label="My Orders"
          aria-labelledby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-8zl1b1-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circular user-avatar css-1mlqxu5-MuiAvatar-root"
          >
            <img
              alt="User-avatar"
              className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="UserAvatar.png"
            />
          </div>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</header>
`;
